modules/experimental-layers/src/utils/gpu-grid-aggregator.js

```diff
@@ -282,104 +508,39 @@ export default class GPUGridAggregator {
       }
       positionsBuffer = new Buffer(gl, {size: 2, data: new Float32Array(positions)});
       weightsBuffer = new Buffer(gl, {size: 1, data: new Float32Array(weights)});
-      this.gridAggregationModel.setAttributes({
+      createPos64xyLow = opts.fp64;
+      Object.assign(aggregationModelAttributes, {
         positions: positionsBuffer,
         weights: weightsBuffer
       });
       this.gridAggregationModel.setVertexCount(positions.length / 2);
       this._setState({positionsBuffer, weightsBuffer});
     }
 
-    if (changeFlags.cellSizeChanged || !gridPixelBuffer) {
-      if (gridPixelBuffer) {
-        gridPixelBuffer.delete();
-      }
-      if (gridTexCoordsBuffer) {
-        gridTexCoordsBuffer.delete();
+    if (createPos64xyLow) {
+      assert(positions64xyLow);
+      if (positions64xyLowBuffer) {
+        positions64xyLowBuffer.delete();
       }
-      const gridPixels = new Float32Array(numCol * numRow * 2);
-      const gridTexCoords = new Float32Array(getTexCoordPerPixel([numCol, numRow]));
-      gridPixelBuffer = new Buffer(gl, {
-        size: 2,
-        data: gridPixels
-      });
-      gridTexCoordsBuffer = new Buffer(gl, {
-        size: 2,
-        data: gridTexCoords
+      positions64xyLowBuffer = new Buffer(gl, {size: 2, data: new Float32Array(positions64xyLow)});
+      Object.assign(aggregationModelAttributes, {
+        positions64xyLow: positions64xyLowBuffer
       });
+      this._setState({positions64xyLowBuffer});
+    }
 
-      this.allAggregationModel.setAttributes({
-        positions: gridPixelBuffer,
-        texCoords: gridTexCoordsBuffer
-      });
-      this.allAggregationModel.setVertexCount(gridPixels.length / 2);
+    this.gridAggregationModel.setAttributes(aggregationModelAttributes);
+
+    if (changeFlags.cellSizeChanged) {
+      this.allAggregationModel.setInstanceCount(numCol * numRow);
 
       const framebufferSize = {width: numCol, height: numRow};
       this.gridAggregationFramebuffer.resize(framebufferSize);
       this.allAggregrationFramebuffer.resize(framebufferSize);
-      this._setState({gridPixelBuffer, gridTexCoordsBuffer});
     }
   }
   /* eslint-enable max-statements */
 
-  _runGridAggregation(opts) {
```

Comment:
Nit: Looks like this code was moved. It is usually better to move code separately in a specially marked first PR, to avoid the diff becoming really big (and making the reviewer re-review the moved but unchanged code).

Reply:
Sure.